custom_components/ha_washdata/store_client.py

```
@@ -241,13 +299,48 @@ async def _run_query(self, sq: dict[str, Any], parent: str = "") -> list[dict[st
+        fut: asyncio.Future[list[dict[str, Any]]] = asyncio.get_event_loop().create_future()
+        self._inflight[key] = fut
+        try:
+            gen = self._cache_gen
+            fetched = await self._run_query(build_sq())
+            result = fetched if fetched is not None else []
+            # Cache only a successful (non-None) and non-superseded read (the generation guard
+            # rejects a snapshot fetched across an invalidation).
+            if fetched is not None and gen == self._cache_gen:
+                self._cache_put(key, result, self._CATALOG_CACHE_TTL_S)
+            fut.set_result(result)
+            return result
```

Ruff (0.15.21)

[warning] Consider moving this statement to an `else` block

(TRY300)